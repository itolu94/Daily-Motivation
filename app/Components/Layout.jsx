import React, {Component} from 'react';
import Header from './Content/Pages/Contact/Header.jsx';
import Registration from './Content/Pages/Registration/Registration.jsx';
import DeleteAccount from './Content/Pages/DeleteAccount/DeleteAccount.jsx';
import Contact from './Content/Pages/Contact/Contact.jsx';
import Footer from './Content/Footer.jsx';
import {LayoutDiv} from "./Layout.style.jsx";

export default class Layout extends Component {
    constructor(){
        super();
        this.state={
            page: ''
        };
        this.changePage =  this.changePage.bind(this);
    }

    changePage(e){
        this.setState({page: e.target.name});
    }

    renderPage(){
        switch(this.state.page){
            case "registration":
                return <Registration/>
            case "delete-account":
                return <DeleteAccount/>
            case "contact":
                return <Contact/>
            default:
                return <Registration />
        }
    }
    render(){
        return(
            <LayoutDiv>
                <Header changePage={this.changePage} />
                {this.renderPage()}
                <Footer/>
            </LayoutDiv>
        )
    }
}