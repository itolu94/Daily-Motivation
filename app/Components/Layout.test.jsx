import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import { shallow } from 'enzyme'
import Layout from './Layout.jsx';
import renderer from 'react-test-renderer';


test("Layout component", () => {
    const wrapper = shallow(<Layout/>)
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({"page": "registration"});
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().page).toEqual("registration");
    wrapper.setState({"page": "delete-account"});
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().page).toEqual("delete-account");
    wrapper.setState({"page": "contact"});
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().page).toEqual("contact");
});
