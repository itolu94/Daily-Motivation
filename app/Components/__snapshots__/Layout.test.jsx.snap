// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component 1`] = `
<styled.div>
  <Header
    changePage={[Function]}
  />
  <Registration />
  <Footer />
</styled.div>
`;

exports[`Layout component 2`] = `
<styled.div>
  <Header
    changePage={[Function]}
  />
  <Registration />
  <Footer />
</styled.div>
`;

exports[`Layout component 3`] = `
<styled.div>
  <Header
    changePage={[Function]}
  />
  <DeleteAccount />
  <Footer />
</styled.div>
`;

exports[`Layout component 4`] = `
<styled.div>
  <Header
    changePage={[Function]}
  />
  <Contact />
  <Footer />
</styled.div>
`;
